tailscale configure kubeconfig panics (nil pointer dereference) when proxy group exists but has no healthy nodes

### What is the issue?

Running tailscale configure kubeconfig <proxy-group-name> causes a nil pointer dereference / SIGSEGV when the named kube-apiserver ProxyGroup is registered in the tailnet but its backing nodes are offline or not yet active.

Instead of returning an error, the process panics and aborts.



### Observed behavior

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10ff43870]

goroutine 17 [running, locked to thread]:
tailscale.com/cmd/tailscale/cli.nodeOrServiceDNSNameFromArg(0x7ee21f1a80d0, 0x7ee21f0a2b60, {0x16b9073ea, 0x11})
      tailscale.com@v1.98.5/cmd/tailscale/cli/configure-kube.go:260 +0x2f0
tailscale.com/cmd/tailscale/cli.runConfigureKubeconfig({0x110cb5390, 0x110dd4020}, {0x7ee21f116020?, 0x1?, 0x7ee21eedeb18?})
      tailscale.com@v1.98.5/cmd/tailscale/cli/configure-kube.go:104 +0x1a0
...
/usr/local/bin/tailscale: line 2: 76575 Abort trap: 6   /Applications/Tailscale.app/Contents/MacOS/tailscale "$@"

### Expected behavior

A clear error message indicating the proxy group has no reachable nodes, e.g.:

error: proxy group "my-apiserver-proxy" has no active nodes visible in the tailnet

### Steps to reproduce

1. Create a kube-apiserver ProxyGroup in a Kubernetes cluster via the Tailscale operator.
2. Ensure the proxy group's pods are not running (e.g. scaled to 0, pending approval in the admin console, or the cluster is unreachable).
3. Run:
tailscale configure kubeconfig <proxy-group-name>

### Are there any recent changes that introduced the issue?

The actual bug is that nodeOrServiceDNSNameFromArg in Tailscale v1.98.x doesn't nil-check before dereferencing the service entry's DNSName field — it should return an error, not segfault. 

### OS

macOS

### OS version

Tahoe

### Tailscale version

1.98.5

### Other software

1.98.5
  tailscale commit: 8f8fe6a2e167459ed0f62616287b61b0b0a54eb5
  long version: 1.98.5-t8f8fe6a2e-gc1619fb10
  other commit: c1619fb10d5db0f7cb1d109d5b67d053f7751508
  go version: go1.26.3 (tailscale/go e877d97384)

### Bug report

_No response_

--- a/cmd/tailscale/cli/configure-kube.go
+++ b/cmd/tailscale/cli/configure-kube.go
@@ -306,6 +306,11 @@ func nodeOrServiceDNSNameFromArg(st *ipnstate.Status, dns *tailcfg.DNSConfig, ar
 	}
 	ipPrefix := netip.PrefixFrom(ip, ip.BitLen())
 	for _, ps := range st.Peer {
+		if ps.AllowedIPs == nil {
+			// Peer with no addresses visible in the tailnet, e.g. a ProxyGroup
+			// whose backing nodes are offline or not yet approved (#20255).
+			continue
+		}
 		for _, allowedIP := range ps.AllowedIPs.All() {
 			if allowedIP == ipPrefix {
 				return rec.Name, nil

--- a/cmd/tailscale/cli/configure-kube_test.go
+++ b/cmd/tailscale/cli/configure-kube_test.go
@@ -7,6 +7,7 @@ package cli
 import (
 	"bytes"
 	"fmt"
+	"net/netip"
 	"os"
 	"path/filepath"
 	"runtime"
@@ -14,6 +15,10 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"tailscale.com/ipn/ipnstate"
+	"tailscale.com/tailcfg"
+	"tailscale.com/types/key"
+	"tailscale.com/types/views"
 )
 
 func TestKubeconfig(t *testing.T) {
@@ -337,5 +342,70 @@ func TestGetInputs(t *testing.T) {
 				})
 			}
 		}
+	}
+}
+
+func TestNodeOrServiceDNSNameFromArg(t *testing.T) {
+	svcIP := netip.MustParseAddr("100.100.100.100")
+	dnsCfg := &tailcfg.DNSConfig{
+		ExtraRecords: []tailcfg.DNSRecord{
+			{Name: "svc.example.ts.net", Value: svcIP.String()},
+		},
+	}
+
+	peerWithService := &ipnstate.PeerStatus{DNSName: "node-a.example.ts.net."}
+	allowed := views.SliceOf([]netip.Prefix{netip.PrefixFrom(svcIP, svcIP.BitLen())})
+	peerWithService.AllowedIPs = &allowed
+
+	// A peer with no AllowedIPs, as reported for a ProxyGroup whose backing
+	// nodes are offline or not yet approved (issue #20255).
+	peerNoAddrs := &ipnstate.PeerStatus{DNSName: "node-b.example.ts.net."}
+
+	tests := []struct {
+		name    string
+		peers   []*ipnstate.PeerStatus
+		arg     string
+		want    string
+		wantErr string
+	}{
+		{
+			name:    "service_with_no_reachable_peer",
+			peers:   []*ipnstate.PeerStatus{peerNoAddrs},
+			arg:     "svc",
+			wantErr: "not currently reachable",
+		},
+		{
+			name:  "service_advertised_by_peer",
+			peers: []*ipnstate.PeerStatus{peerNoAddrs, peerWithService},
+			arg:   "svc",
+			want:  "svc.example.ts.net",
+		},
+		{
+			name:  "node_dns_name",
+			peers: []*ipnstate.PeerStatus{peerNoAddrs},
+			arg:   "node-b",
+			want:  "node-b.example.ts.net.",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			st := &ipnstate.Status{Peer: map[key.NodePublic]*ipnstate.PeerStatus{}}
+			for _, ps := range tt.peers {
+				st.Peer[key.NewNode().Public()] = ps
+			}
+			got, err := nodeOrServiceDNSNameFromArg(st, dnsCfg, tt.arg)
+			if tt.wantErr != "" {
+				if err == nil || !strings.Contains(err.Error(), tt.wantErr) {
+					t.Fatalf("err = %v, want error containing %q", err, tt.wantErr)
+				}
+				return
+			}
+			if err != nil {
+				t.Fatal(err)
+			}
+			if got != tt.want {
+				t.Errorf("got %q, want %q", got, tt.want)
+			}
+		})
 	}
 }

